lld/ELF/Arch/LoongArch.cpp

```diff
@@ -830,6 +833,45 @@ static void relaxPCHi20Lo12(Ctx &ctx, const InputSection &sec, size_t i,
   remove = 4;
 }
 
+static bool isInsnPairCall36(uint64_t pair) {
+  const uint32_t insn1 = extractBits(pair, 31, 0);
+  const uint32_t insn2 = extractBits(pair, 63, 32);
+  if ((insn1 & 0xfe000000) != PCADDU18I)
```

Comment:
There is an instruction that #123576 doesn't have. We typically avoid instruction checks unless absolutely necessary.

Reply:
`ld.bfd` checks for `jirl` but not `pcaddu18i`, nor does `ld.bfd` ensure `jirl.rj == pcaddu18i.rd`. Granted, these are irrelevant for well-formed inputs, but I think such checks should get added to `ld.bfd` (or indeed, the psABI) as well, for the guarantee that "relaxation does not affect code semantics" to uphold.